src/workspace-data/data-table/wds/SingleEntityEditorWds.ts

```diff
@@ -0,0 +1,105 @@
+import _ from 'lodash/fp';
+import { Fragment, useState } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { ButtonPrimary, ButtonSecondary, spinnerOverlay } from 'src/components/common';
+import Modal from 'src/components/Modal';
+import { reportError } from 'src/libs/error';
+
+import { getAttributeType } from './attribute-utils';
+import AttributeInput from './AttributeInput';
+
+export interface SingleEntityWdsProps {
+  recordType: string;
+  recordName: string;
+  attributeName: string;
+  attributeValue: any;
+  workspaceId: string;
+  onDismiss: any;
+  onSuccess: any;
+  dataProvider: any;
+  recordTypeAttributes: { name: string; datatype: string }[];
+}
+
+export const SingleEntityEditorWds = ({
+  recordType,
+  recordName,
+  attributeName,
+  attributeValue,
+  workspaceId,
+  onDismiss,
+  onSuccess,
+  dataProvider,
+  recordTypeAttributes,
+}: SingleEntityWdsProps) => {
+  const { type: originalValueType } = getAttributeType(attributeName, recordTypeAttributes, dataProvider);
+  const [newValue, setNewValue] = useState(attributeValue);
+  const isUnchanged = _.isEqual(attributeValue, newValue);
+
+  const [isBusy, setIsBusy] = useState(false);
+
+  const modalTitle = `Edit ${attributeName} value for ${recordName}`;
+  const modalDetails = originalValueType ? `Column Type is  ${originalValueType}` : '';
```

Comment:
Since the title is sentence case, suggest sentence case here as well.

```suggestion
  const modalDetails = originalValueType ? `Column type is ${originalValueType}` : '';
```